File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.util.*;
class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n=sc.nextInt();
        sc.nextLine();
        int cnt=0;
        for(int i=0; i<n; i++){
            String str=sc.nextLine();
            Set<Character> set=new HashSet<>();
            char ch='0';
            boolean sign=true;
            for(int j=0; j<str.length(); j++){
                if(ch!=str.charAt(j)){
                    if(set.contains(str.charAt(j))){
                        sign=false;
                        break;
                    }
                    else{
                        ch=str.charAt(j);
                        set.add(ch);
                    }
                }
            }
            if(sign) cnt++;
        }
        System.out.println(cnt);

    }
}

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.cs
using System;
using System.Collections.Generic;

class Program
{
    public static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        int result = 0;
        for (int i = 0; i < n; i++)
        {
            var str = Console.ReadLine();
            HashSet<char> set = new HashSet<char>();
            set.Add(str[0]);
            bool sign = false;
            for (int j = 1; j < str.Length; j++)
            {
                if (str[j] == str[j - 1]) continue;
                else
                {
                    if (set.Contains(str[j]))
                    {
                        sign = true;
                        break;
                    }
                    else set.Add(str[j]);
                }
            }

            if (!sign) result++;

        }

        Console.WriteLine(result);
    }
}